### PR TITLE
fix(tracing): add Context back to the public api

### PR DESCRIPTION
**Pull request by mabdinur**
## Description

Resolves: https://github.com/DataDog/dd-trace-py/issues/9492

Importing `Context` from `ddtrace.context` raises a deprecation warning but this class is required to tracing across execution boundaries (ex: [distributed tracing](https://ddtrace.readthedocs.io/en/v2.9.4/advanced_usage.html#disttracing), [tracing across threads](https://ddtrace.readthedocs.io/en/v2.9.4/advanced_usage.html#tracing-across-threads)). This class can not be removed from the public API without providing users with a different mechanism for tracing across execution boundaries. Although users are not expected to directly initialize this class, the Context object is used by the public api. Therefore we need to expose this type to users.

Since `ddtrace.context` module is deprecated, the `Context` class is now exposed in `ddtrace.__init__.py`. 

## TODO

- The tracing public api is exposed in `ddtrace.__init__.py`. `ddtrace` is not just tracing library, it is a datadog sdk. We should define and expose the tracing public api in a `ddtrace/trace/` directory.
- Expose Span and Tracer types in the public api.

## 

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by taegyunkim — APPROVED** [review]

File: ddtrace/opentelemetry/_trace.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def start_span(
         curr_otel_span = get_current_span(context)
         if curr_otel_span is INVALID_SPAN:
             # There is no active datadog/otel span
-            dd_active = None  # type: Optional[Union[ddtrace.context.Context, ddtrace.Span]]
+            dd_active = None  # type: Optional[Union[ddtrace.Context, ddtrace.Span]]
```
Comment (mabdinur) [inline ddtrace/opentelemetry/_trace.py:105]:
```suggestion
            dd_active = None  # type: Optional[Union[ddtrace.trace.Context, ddtrace.Span]]
```

File: releasenotes/notes/add-context-back-to-public-api-e6c2412165df10a7.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    tracing: Adds ``ddtrace.trace.Context`` to the public api. This class can now be used to propagate context across execution boundries (ex: threads).
```
Comment (mabdinur) [inline releasenotes/notes/add-context-back-to-public-api-e6c2412165df10a7.yaml:4]:
```suggestion
    tracing: Adds ``ddtrace.trace.Context`` to the public api. This class can now be used to propagate context across execution boundaries (ex: threads).
```

**Review by juanjux — APPROVED** [review]

**Review by quinna-h — APPROVED** [review]

**Review by erikayasuda — APPROVED** [review]

**Comment by brettlangdon** [comment]
@mabdinur there appears to be a conflict that needs to be resolved.

File: ddtrace/context.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 
 deprecate(
-    "The context interface is deprecated.",
-    message="The trace context is an internal interface and will no longer be supported.",
+    "The ddtrace.context module is deprecated and will be removed from the public API.",
+    message="Context class should be imported from ddtrace.trace package",
```
Comment (mabdinur) [inline ddtrace/context.py:8]:
```suggestion
    message="Context should be imported from ddtrace.trace package",
```

**Pull request merged**